rclpy/rclpy/publisher.py

```diff
@@ -64,10 +64,13 @@ def publish(self, msg: MsgType) -> None:
         :raises: TypeError if the type of the passed message isn't an instance
           of the provided type when the publisher was constructed.
         """
-        if not isinstance(msg, self.msg_type):
-            raise TypeError()
         with self.handle as capsule:
-            _rclpy.rclpy_publish(capsule, msg)
+            if isinstance(msg, self.msg_type):
+                _rclpy.rclpy_publish(capsule, msg)
+            elif isinstance(msg, bytes):
```

Comment:
Also update the type annotation and parameter description to reflect this change:

```python
def publish(self, msg: Union[MsgType, bytes]) -> None:
```